agent/dns.go

```diff
@@ -413,7 +413,8 @@ func (d *DNSServer) handlePtr(resp dns.ResponseWriter, req *dns.Msg) {
 	m.SetReply(req)
 	m.Compress = !cfg.DisableCompression
 	m.Authoritative = true
-	m.RecursionAvailable = (len(cfg.Recursors) > 0)
+	recursionAvailable := len(cfg.Recursors) > 0 && atomic.LoadUint32(&(d.recursorEnabled)) != 1
```

Comment:
micro-nit: I think we only need one check or the other since the logic is the same for setting `recursorEnabled` in `toggleRecursorHandlerFromConfig`. I guess `recursorEnabled` is safer because the handler is guaranteed.

Reply:
This is fixed in 1e85c5c070032853805321c551e1dfc74be57a86